.azure-pipelines/Linux-CI.yml

```diff
@@ -108,14 +108,21 @@ jobs:
       source venv/bin/activate
       python onnx/backend/test/cmd_tools.py generate-data --clean
       git status
+      if [ '$(onnx_ml)' == '1' ]; then
+        export exclude_onn_ml_filter=''
+      else
+        export exclude_onn_ml_filter=':!*/data/node/test_ai_onnx_ml_*'
```

Comment:
May I understand why we would like to ignore the added test directory (`/data/node/test_ai_onnx_ml_*'`) entirely? Typically we only exclude output_*.pb, because np.random will possibly produce different result from different platforms. I thought the model itself should always be consistent?

Reply:
test named with test_ai_onnx_ml_*  (will put into a folder as @xadupre suggested one this code passes) can only get generated when ONNX_ML is set. So we need to exclude git diff test on these files is ONNX_ML is not set because they will be shown as deleted instead of unchanged.

Reply:
Thank you for the context. If that is the case, I am inclined to change the logic in the generate-data tool instead. For instance, do not delete those directories with "test_ai_onnx_ml_*" prefix when ONNX_ML=0. It is one behavior change in tool vs a few modifications in relevant pipelines.

Reply:
good point! @jcwchen it makes this PR much cleaner.